### PR TITLE
Updating code owners of "docs/"

### PR DESCRIPTION
**Pull request by bozturkMSFT**
This PR adds Kathleen as the owner of the docs. This means that whenever one of the files under docs folder change, her review will be automatically requested.

### CODE REVIEW

**Review by donJoseLuis — APPROVED** [review]

File: CODEOWNERS
Comment (vlada-shubina) [inline CODEOWNERS:2]: @bozturkMSFT 
- can we also add template-engine-maintainers here? We need two reviews from them as well as per current settings. 
- should it be '/docs/?

Also I think it's nice to make the following changes to the rules
- in case doc owner haven't approved the PR, merging is blocked. 
However, open for debate here.

**Review by vlada-shubina — CHANGES_REQUESTED** [review]

File: CODEOWNERS
Comment (AR-May) [reply CODEOWNERS:2]: I do not think we need the last rule. We still may have changes that do not need PM approval. For example, adding a template to the "Available templates" wiki page. There is no need to block merging in this case.

**Review by grinrag — APPROVED** [review]

File: CODEOWNERS
Comment (bozturkMSFT) [reply CODEOWNERS]: Added the maintainers team to the owners of docs.
Updated to `/docs/` as the previous value meant "any docs folder under any directory".
I had already enabled requiring a review from code owners. For changes that don't need PM approval, we can just override the approval requirement since almost all the team has admin rights.

**Review by vlada-shubina — APPROVED** [review]

**Review by AR-May — APPROVED** [review]

**Pull request merged**
